### PR TITLE
Remove install path dependency from Linux builds

### PR DESCRIPTION
**Pull request by daschuer**
This is required because the install path is changed by cpack -G deb after building the binaries. 

Now it creates the install prefix from the binary location. 

For debugging, it holds now the cmake bin and source directory to find the resources without copying them around.
This has already caused some issues on windows, because there we cant use a symlink.

### CODE REVIEW

**Comment by Holzhaus** [comment]
> For debugging, it holds now the cmake bin and source directory to find the resources without copying them around.
This has already caused some issues on windows, because there we cant use a symlink.

Please don't. I think @Be-ing was already starting to work on reproducible builds and this is a step in the wrong direction.

File: src/preferences/configobject.cpp [hunk 1]
```diff
@@ -20,36 +20,28 @@ QString computeResourcePath() {
...
-#ifdef __WINDOWS__
+#elif __WINDOWS__
         // On Windows, set the config dir relative to the application dir if all
         // of the above fail.
         else {
             qResourcePath = QCoreApplication::applicationDirPath();
         }
-#endif
-#ifdef __APPLE__
+#elif __APPLE__
```
Comment (Holzhaus) [inline src/preferences/configobject.cpp:44]: This is always false. Same for `#elif __WINDOWS__` above. 

```suggestion
#elif defined(__APPLE_) 
```

**Comment by daschuer** [comment]
OK, I have changed the approach. Now Mixxx watches out for a CMakeCache.txt file and when its found, it reads the source path from there.

**Comment by Be-ing** [comment]
Clever. I like that approach.

File: src/preferences/configobject.cpp [hunk 2]
```diff
@@ -14,42 +14,44 @@
...
+            DEBUG_ASSERT(QDir(qResourcePath).exists());
         }
-#ifdef __UNIX__
-        // On Linux if all of the above fail the /usr/share path is the logical
-        // place to look.
-        else {
-            qResourcePath = UNIX_SHARE_PATH;
+#if defined(__UNIX__)
+        else if (mixxxDir.cdUp() && mixxxDir.cd(QStringLiteral("/share/mixxx"))) {
+            qResourcePath = mixxxDir.absolutePath();
```
Comment (Be-ing) [inline src/preferences/configobject.cpp:46]: I don't think this should be platform dependent. Wouldn't this work on Windows and macOS too?

**Comment by Be-ing** [comment]
I tested and confirm this new method still works on Linux.

File: src/preferences/configobject.cpp (hunk 2 above)
Comment (daschuer) [reply src/preferences/configobject.cpp:46]: Only Linux (Unix) has the 
..Install Prefix 
 /bin
 /share 
   /mixxx  
Layout 
So this is correct here.

**Comment by daschuer** [comment]
> I tested and confirm this new method still works on Linux.

OK, merge then?

**Pull request merged**